biom/tests/test_table.py

```diff
@@ -1016,64 +1016,72 @@ def test_to_from_hdf5_bug_861(self):
                   ['c', 'd', 'e'])
         t.add_metadata({'a': {'a / problem': 10}, 'b': {'a / problem': 20}},
                        axis='observation')
-        with NamedTemporaryFile() as tmpfile:
+        with NamedTemporaryFile(delete=False) as tmpfile:
             h5 = h5py.File(tmpfile.name, 'w')
             t.to_hdf5(h5, 'tests')
             h5.close()
 
             h5 = h5py.File(tmpfile.name, 'r')
             obs = Table.from_hdf5(h5)
+            h5.close()
+        os.unlink(tmpfile.name)
```

Comment:
same here

Reply:
These tests are creating the temporary files outside of a `setUp`/`tearDown` context, so the solution involving putting the `unlink` in `tearDown` won't work here. The reason the `close`s are necessary within the context manager is that without them `unlink` will throw ` PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\me\\AppData\\Local\\Temp\\tmpg4pk3ink'` . My understanding is that the context manager will close `tmpfile` when it is finished, but it won't close `h5`. Exactly why this is the case I'm unsure of.

Reply:
I think this is the case because I set `delete=False` during the file creation, so a file will not be deleted when it is closed. But it seems on windows that file handles must be closed before the file may be deleted.

Reply:
So this solution works because it closes all file handles, including `tmpfile` and `h5`, and then deletes the file using `unlink`. I'm not sure of a better solution for this particular case and the other similar cases.

Reply:
Issue is edge case if test excepts prior to `unlink`. A common strategy for this type of stuff:

```python
def setUp(self):
    self.temp_files = []
def tearDown(self):
    for f in self.temp_files:
        os.unlink(f)
def test_some_test(self):
    ...
    self.temp_files.append(blah)
    ...
```